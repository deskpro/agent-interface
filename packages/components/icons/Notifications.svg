<svg width="25" height="24" viewBox="0 0 25 24" fill="none" xmlns="http://www.w3.org/2000/svg">

<path d="M18.3286 12.8381V8.22857C18.3286 5.81905 16.7571 3.82857 14.6619 3.09524V2.78095C14.6619 1.73333 13.8238 1 12.881 1C11.8333 1 10.8905 1.73333 10.8905 2.78095V3.09524C8.79524 3.82857 7.22381 5.92381 7.22381 8.22857V12.8381L4.5 16.5048V19.3333H21.0524V16.5048L18.3286 12.8381ZM19.1667 17.5524H6.38571V17.1333L8.79524 13.9905L9.21429 13.4667V12.8381V8.22857C9.21429 6.44762 10.4714 4.98095 12.1476 4.66667H13.6143C15.2905 4.98095 16.5476 6.44762 16.5476 8.22857V12.8381V13.4667L16.9667 13.9905L19.3762 17.1333V17.5524H19.1667Z" />
<path d="M12.7762 23C14.1381 23 15.5 22.1619 15.5 21.2191H10.0524C10.0524 22.1619 11.4143 23 12.7762 23Z" />

</svg>
