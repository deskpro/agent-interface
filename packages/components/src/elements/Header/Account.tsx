import * as React from "react";
import { Reference, Popper, Manager } from "react-popper";
import Arrow from "../../Button/Arrow";
import Menu from "../Menu/Menu";

const Account: React.FC = () => {
  const [isOpen, setOpenState] = React.useState<boolean>(false);
  return (
    <Manager>
      <Reference>
        {({ ref }) => (
          <span className="dp-Account" ref={ref}>
            <span className="dp-Icon dp-IconPlane dp-ActionDropDown" />
            <Arrow isActive={isOpen} onClick={() => setOpenState(!isOpen)} />
          </span>
        )}
      </Reference>
      {isOpen && (
        <Popper
          placement="bottom-end"
          modifiers={{ offset: { offset: "0,15" } }}
        >
          {({ ref, style }) => (
            <Menu
              className="Menu--tick dp-YourAccount"
              title="Your account"
              menuRef={ref}
              style={style}
            >
              <Menu.MenuItem
                icon={<span className="dp-Icon dp-IconPlane" />}
                name="raven"
                text="Raven Paper"
              />
              <Menu.MenuItem
                icon={<span className="dp-Icon dp-IconPlaneOrange" />}
                name="raven-craft"
                text="Raven Paper Crafts"
              />
              <Menu.MenuItem
                icon={<span className="dp-Icon dp-IconPlaneViolet" />}
                name="raven-eu"
                text="RP Europe (Demo)"
              />
              <Menu.MenuItem icon="add-tab" name="add-tab" text="Add new" />
              <Menu.MenuItem
                icon={<span className="dp-Icon dp-IconNoteGray" />}
                name="knowledge"
                text="Knowledgebase article"
              />
            </Menu>
          )}
        </Popper>
      )}
    </Manager>
  );
};

export default Account;
