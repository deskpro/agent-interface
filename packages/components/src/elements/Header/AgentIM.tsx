import * as React from "react";
import Icon from "../Icon/Icon";

import "@deskpro/agent-interface-style/dist/elements/dp-badges.css";

const AgentIM: React.FC = () => (
  <div>
    <ul className="dp-Agent-IM">
      <li className="dp-MenuBarItem">
        <span className="dp-Badge Badge--round Badge--grey-lighter" />
      </li>
      <li className="dp-MenuBarItem">
        <span className="dp-Badge Badge--round Badge--grey-lighter" />
      </li>
      <li className="dp-MenuBarItem">
        <span className="dp-Badge Badge--round Badge--grey-lighter">1</span>
      </li>
      <li className="dp-MenuBarItem">
        <Icon name="add-im" size={25} />
      </li>
    </ul>
  </div>
);

export default AgentIM;
