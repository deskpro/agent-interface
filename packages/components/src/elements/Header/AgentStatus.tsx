import * as React from "react";
import Icon from "../Icon/Icon";
import Arrow from "../../Button/Arrow";

const AgentStatus: React.FC = () => (
  <div>
    <ul className="dp-Agent-Status">
      <li className="dp-MenuBarItem">
        <Icon name="voice-call" color="success" size={20} />
      </li>
      <li className="dp-MenuBarItem">
        <Icon name="chat" size={22} />
      </li>
      <li className="dp-MenuBarItem">
        <span className="dp-Badge Badge--round Badge--grey-lighter" />
        <Arrow />
      </li>
    </ul>
  </div>
);

export default AgentStatus;
