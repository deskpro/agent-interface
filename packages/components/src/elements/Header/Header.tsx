import * as React from "react";
import classNames from "classnames";
import { CSSTransition } from "react-transition-group";

import Icon from "../Icon/Icon";
import "@deskpro/agent-interface-style/dist/components/dp-Header.css";

export type HeaderProps = {
  className?: string;
  searchKeyword?: string;
  onSearch: (keyword: string) => void;
  left: React.ReactNode;
  right: React.ReactNode;
};

const Header: React.FC<HeaderProps> = ({
  className,
  searchKeyword = "",
  onSearch,
  left,
  right
}) => {
  const [keyword, updateKeyword] = React.useState<string>(searchKeyword);
  React.useEffect(() => {
    updateKeyword(searchKeyword);
  }, [searchKeyword]);

  const [isSearchActive, setSearchState] = React.useState<boolean>(
    !!searchKeyword
  );

  const handleSearchSubmit = React.useCallback(
    (e: React.SyntheticEvent) => {
      e.preventDefault();
      onSearch(keyword);
    },
    [keyword, onSearch]
  );

  const handleSearchEdit = React.useCallback(
    e => {
      if (e.key === "Enter" || e.key === "Tab") {
        handleSearchSubmit(e);
      }
      updateKeyword(e.target.value);
    },
    [updateKeyword, handleSearchSubmit]
  );

  return (
    <div className={classNames("dp-Header", className)}>
      <CSSTransition in={isSearchActive} classNames="search" timeout={500}>
        <div className="dp-SearchBox">
          <form className="dp-SearchForm" onSubmit={handleSearchSubmit}>
            <input
              type="text"
              placeholder="Search..."
              value={keyword}
              onChange={handleSearchEdit}
              onFocus={() => setSearchState(true)}
              onBlur={() => {
                if (!keyword) {
                  setSearchState(false);
                }
              }}
            />
            <Icon name="find" size={18} />
          </form>
          {isSearchActive && (
            <ul className="dp-IconPanelList">
              <li className="dp-MenuBarItem">
                <Icon name="star" size={18} />
              </li>
              <li className="dp-MenuBarItem dp-Close">
                <Icon
                  name="close"
                  size={12}
                  onClick={() => setSearchState(false)}
                />
              </li>
            </ul>
          )}
        </div>
      </CSSTransition>
      <CSSTransition in={!isSearchActive} classNames="fade" timeout={500}>
        <div className="dp-HeaderControls">{left}</div>
      </CSSTransition>
      <div className="dp-HeaderControls is-right">{right}</div>
    </div>
  );
};

export default Header;
