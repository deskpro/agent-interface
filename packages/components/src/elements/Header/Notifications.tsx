import * as React from "react";
import Icon from "../Icon/Icon";

const Notifications: React.FC = () => (
  <div className="dp-Notifications">
    <Icon name="notifications" size={21} />
  </div>
);

export default Notifications;
