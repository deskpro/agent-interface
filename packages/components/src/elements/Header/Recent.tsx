import * as React from "react";
import Icon from "../Icon/Icon";

const Recent: React.FC = () => (
  <div className="dp-Recent">
    <Icon name="history" size={25} />
  </div>
);

export default Recent;
