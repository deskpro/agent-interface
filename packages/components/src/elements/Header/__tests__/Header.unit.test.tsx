import * as React from "react";
import { render } from "react-testing-library";

import Header from "../Header";
import Recent from "../Recent";
import Account from "../Account";
import AgentIM from "../AgentIM";
import AgentStatus from "../AgentStatus";
import Notifications from "../Notifications";

describe("<Header />", () => {
  it("should match its snapshot", () => {
    const { container } = render(
      <Header
        onSearch={jest.fn()}
        left={
          <>
            <Recent />
            <AgentIM />
          </>
        }
        right={
          <>
            <Notifications />
            <AgentStatus />
            <Account />
          </>
        }
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
