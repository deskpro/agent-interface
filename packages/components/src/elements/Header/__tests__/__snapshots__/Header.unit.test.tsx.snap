// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match its snapshot 1`] = `
<div
  class="dp-Header"
>
  <div
    class="dp-SearchBox"
  >
    <form
      class="dp-SearchForm"
    >
      <input
        placeholder="Search..."
        type="text"
        value=""
      />
      <span
        class="dp-Icon Icon--find"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="18"
          width="18"
        />
      </span>
    </form>
  </div>
  <div
    class="dp-HeaderControls"
  >
    <div
      class="dp-Recent"
    >
      <span
        class="dp-Icon Icon--history"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="25"
          width="25"
        />
      </span>
    </div>
    <div>
      <ul
        class="dp-Agent-IM"
      >
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Badge Badge--round Badge--grey-lighter"
          />
        </li>
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Badge Badge--round Badge--grey-lighter"
          />
        </li>
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Badge Badge--round Badge--grey-lighter"
          >
            1
          </span>
        </li>
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Icon Icon--add-im"
            role="button"
            tabindex="-1"
          >
            <icon-mock
              height="25"
              width="25"
            />
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="dp-HeaderControls is-right"
  >
    <div
      class="dp-Notifications"
    >
      <span
        class="dp-Icon Icon--notifications"
        role="button"
        tabindex="-1"
      >
        <icon-mock
          height="21"
          width="21"
        />
      </span>
    </div>
    <div>
      <ul
        class="dp-Agent-Status"
      >
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Icon Icon--voice-call Icon--success"
            role="button"
            tabindex="-1"
          >
            <icon-mock
              height="20"
              width="20"
            />
          </span>
        </li>
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Icon Icon--chat"
            role="button"
            tabindex="-1"
          >
            <icon-mock
              height="22"
              width="22"
            />
          </span>
        </li>
        <li
          class="dp-MenuBarItem"
        >
          <span
            class="dp-Badge Badge--round Badge--grey-lighter"
          />
          <span
            class="dp-Icon dp-Arrow"
            role="button"
            tabindex="-1"
          />
        </li>
      </ul>
    </div>
    <span
      class="dp-Account"
    >
      <span
        class="dp-Icon dp-IconPlane dp-ActionDropDown"
      />
      <span
        class="dp-Icon dp-Arrow"
        role="button"
        tabindex="-1"
      />
    </span>
  </div>
</div>
`;
